### DISCUSSION TITLE
无法禁言匿名用户，因为群消息获得的anonymous的flag为空，id也为空

### DISCUSSION THREAD

**Discussion by anaivebird** [howmanybots/onebot#141] [问答]
无法通过API，set_group_anonymous_ban来禁言匿名用户
因为该API需要anonymous_flag
这个flag只能通过监听群消息来获得

而获得的群消息anonymous.flag为空，anonymous.id也为空
{'anonymous': {'flag': '', 'id': 0, 'name': '嫦娥'}, 'font': 0, 'group_id': 963425306, 'message': [MessageSegment(type='text', data={'text': 'asdf'})], 'message_id': 98064478, 'message_type': 'group', 'post_type': 'message', 'raw_message': 'asdf', 'self_id': 我的qq号, 'sender': {'age': 0, 'area': '', 'level': '', 'nickname': '匿名消息', 'sex': 'unknown', 'user_id': 80000000}, 'sub_type': 'anonymous', 'time': 1608231870, 'user_id': 80000000}

**Comment by ExerciseBook** [comment]
您好，onebot 仅为一个协议的规范，不能够知晓您所使用的 onebot 实现是否成功实现了这部分的功能。建议您到您所使用的插件的 repo 下进行询问。

如果您不知道您使用的是哪一个插件，您可以将其的基本信息描述出来让热心网友一同帮您解决。

**Comment by anaivebird** [comment]
> 您好，onebot 仅为一个协议的规范，不能够知晓您所使用的 onebot 实现是否成功实现了这部分的功能。建议您到您所使用的插件的 repo 下进行询问。
> 
> 如果您不知道您使用的是哪一个插件，您可以将其的基本信息描述出来让热心网友一同帮您解决。

您好，谢谢您的回复，现在市面上mirai系的插件似乎都不支持，比如go-cqhttp是基于Mirai-GO的，就不支持，见https://github.com/Mrs4s/go-cqhttp/issues/504

**Comment by ExerciseBook** [comment]
那可能是 mirai 真的没有这个接口了？

**Comment by anaivebird** [comment]
> 那可能是 mirai 真的没有这个接口了？

您好，刚才找到mirai的issue了，https://github.com/mamoe/mirai/issues/432
属于unplanned feature
